tt_eager/tensor/tensor_impl.hpp

```diff
@@ -385,7 +385,6 @@ inline Tensor to_host(const Tensor& tensor, bool blocking = true) {
             host_tensor.set_dtype(tensor.get_dtype());
             host_tensor.set_layout(tensor.get_layout());
             insert_buffer_and_shape_for_device(device, shard, host_tensor, device_index);
-            host_tensor.set_populated(device);
```

Comment:
why is this no longer needed?

Reply:
This function no longer exists. Instead, we check the number of workers that have completed and compare them against the number of shards in the tensor. `tensor.cpu()` which eventually calls this function does that handling.